apollo-ios/Sources/Apollo/GraphQLResult.swift

```diff
@@ -76,3 +128,48 @@ extension GraphQLResult {
       return val
   }
 }
+
+fileprivate extension DataDict {
+  enum Error: Swift.Error, LocalizedError {
+    case invalidPathDataType(String)
+    case cannotOverwriteData(AnyHashable, AnyHashable)
+
+    public var errorDescription: String? {
+      switch self {
+      case let .invalidPathDataType(got):
+        return "Invalid data type for merge - \(got)"
+
+      case let .cannotOverwriteData(current, new):
+        return "Incremental merging cannot overwrite existing data \(current) with \(new)"
```

Comment:
Lets make this error message more clear. Indicated that increment response had mismatching value for an existing field.

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/6d737393cc3b85b54140171e3828574ce1832268.